Source/Core/DolphinQt2/Settings/GameCubePane.cpp

```diff
@@ -0,0 +1,333 @@
+
+  m_skip_main_menu = new QCheckBox(tr("Skip Main Menu"));
+  m_override_language_ntsc = new QCheckBox(tr("Override Language on NTSC Games"));
+  m_language_combo = new QComboBox;
+
+  // Add languages
+  for (const auto& language :
+       {tr("English"), tr("German"), tr("French"), tr("Spanish"), tr("Italian"), tr("Dutch")})
+    m_language_combo->addItem(language);
+
+  ipl_layout->addWidget(m_skip_main_menu, 0, 0);
+  ipl_layout->addWidget(new QLabel(tr("System Language:")), 1, 0);
+  ipl_layout->addWidget(m_language_combo, 1, 1);
+  ipl_layout->addWidget(m_override_language_ntsc, 2, 0);
+
+  // Device Settings
+  QGroupBox* device_box = new QGroupBox(tr("Device Settings"));
+  QGridLayout* device_layout = new QGridLayout;
+  device_box->setLayout(device_layout);
+
+  m_slot_combos[0] = new QComboBox;
+  m_slot_combos[1] = new QComboBox;
+  m_slot_combos[2] = new QComboBox;
+
+  m_slot_buttons[0] = new QPushButton(tr("..."));
+  m_slot_buttons[1] = new QPushButton(tr("..."));
+
+  m_slot_buttons[0]->setSizePolicy(QSizePolicy::Fixed, QSizePolicy::Fixed);
+  m_slot_buttons[1]->setSizePolicy(QSizePolicy::Fixed, QSizePolicy::Fixed);
+
+  const QString i10n_nothing = tr("<Nothing>");
+  const QString i10n_dummy = tr("Dummy");
+
+  // Add slot devices
+
+  for (const auto& device : {i10n_nothing, i10n_dummy, tr("Memory Card"), tr("GCI Folder"),
+                             tr("USB Gecko"), tr("Advance Game Port"), tr("Microphone")})
+  {
+    m_slot_combos[0]->addItem(device);
+    m_slot_combos[1]->addItem(device);
+  }
+
+  // Add SP1 devices
+
+  for (const auto& device : {i10n_nothing, i10n_dummy, tr("Broadband Adapter")})
+  {
+    m_slot_combos[2]->addItem(device);
+  }
+
+  device_layout->addWidget(new QLabel(tr("Slot A:")), 0, 0);
+  device_layout->addWidget(m_slot_combos[0], 0, 1);
+  device_layout->addWidget(m_slot_buttons[0], 0, 2);
+  device_layout->addWidget(new QLabel(tr("Slot B:")), 1, 0);
+  device_layout->addWidget(m_slot_combos[1], 1, 1);
+  device_layout->addWidget(m_slot_buttons[1], 1, 2);
+  device_layout->addWidget(new QLabel(tr("SP1:")), 2, 0);
+  device_layout->addWidget(m_slot_combos[2], 2, 1);
+
+  layout->addWidget(ipl_box);
+  layout->addWidget(device_box);
+
+  setLayout(layout);
+}
+
+void GameCubePane::ConnectWidgets()
+{
+  // IPL Settings
+  connect(m_skip_main_menu, &QCheckBox::stateChanged, this, &GameCubePane::SaveSettings);
+  connect(m_language_combo, static_cast<void (QComboBox::*)(int)>(&QComboBox::currentIndexChanged),
+          this, &GameCubePane::SaveSettings);
+  connect(m_override_language_ntsc, &QCheckBox::stateChanged, this, &GameCubePane::SaveSettings);
+
+  // Device Settings
+  for (int i = 0; i < 3; i++)
+  {
+    connect(m_slot_combos[i],
+            static_cast<void (QComboBox::*)(int)>(&QComboBox::currentIndexChanged), this,
+            &GameCubePane::SaveSettings);
+    if (i < 2)
+    {
+      connect(m_slot_buttons[i], &QPushButton::pressed, this, [this, i] { OnConfigPressed(i); });
```

Comment:
Any chance you could remove some of these magic numbers? It feels a bit error prone...

Would it be possible to change OnConfigPressed to take a pointer/reference to a m_slot_combos item directly?